### PR TITLE
Reset observer on navigator recreation

### LINKED ISSUE #4557: Add navigator observer on recreation event
When native navigator is recreated we need to reset observer to receive updates from a new navigator instance.

### PR DESCRIPTION
**Pull request by korshaknn**
### Description

closes #4557

### Changelog

```
<changelog>Fixed a bug where in some cases status updates were not generated after switching between offline and online states.</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt [hunk 1]
```diff
@@ -161,6 +161,7 @@ internal class MapboxTripSession(
             if (fallbackVersionsObservers.isNotEmpty()) {
                 navigator.setFallbackVersionsObserver(nativeFallbackVersionsObserver)
             }
+            navigator.addNavigatorObserver(navigatorObserver)
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:164]: scenario: 
- `MapboxNavigation` is created
- navigator switched to offline/online more
- start TripSession

in this case, `navigatorObserver` is set to navigator twice. Is it expected? should we check if a session is started?

**Review by LukasPaczos — COMMENTED** [review]
Could you add a unit test?

Also changelog suggestion to explain what has changed for the user, rather than how it looks internally. Something like:
```
Fixed a bug where status updates were not generated after switching between offline and online states.
```

Is this message correct?

**Comment by korshaknn** [comment]
@RingerJK @LukasPaczos ptal

**Review by LukasPaczos — APPROVED** [review]
LGTM

**Review by RingerJK — APPROVED** [review]

**Pull request merged**